mlir/utils/performance/tuningRunner.py

```diff
@@ -0,0 +1,196 @@
+    """
+    if args is None:
+        args = sys.argv[1:]
+
+    archNames = perfRunner.getArch()
+    arch = ','.join(archNames)
+
+    root_dir = str(subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).decode().strip())
+    default_conv_configs = root_dir + '/mlir/utils/jenkins/miopen-tests/resnet50-miopen-configs'
+
+    parser = argparse.ArgumentParser(
+        prog="rocMLIR tuning runner",
+        description="A script for tunning MLIR conv2d or gemm kernels",
+        allow_abbrev=False,
+    )
+
+    parser.add_argument("--op", "--operation", choices=['conv', 'gemm'],
+        default='conv',
+        help="Operation for tuning")
+
+    mutex_arg_group = parser.add_mutually_exclusive_group()
+    parser.add_argument(
+        "-c", "--configs_file",
+        type=str,
+        default=default_conv_configs,
+        help="File of configurations to test"
+    )
+
+    parser.add_argument(
+        "--mlir-build-dir",
+        type=str,
+        default=perfRunner.find_mlir_build_dir(),
+        help="The build directory of MLIR based kernel generator",
+    )
+
+    parser.add_argument(
+        "--miopen-build-dir",
+        type=str,
+        default=perfRunner.find_miopen_build_dir(),
+        help="The build directory of MIOpen",
+    )
+
+    parser.add_argument(
+        "--config",
+        type=str,
+        nargs='*',
+        help="The specific config to test, if you want to test one"
+    )
+
+    parser.add_argument(
+        "--rocmlir_gen_flags",
+        type=str,
+        default=argparse.SUPPRESS,
+        help="rocmlir-gen flags to toggle each feature"
+    )
+
+    parser.add_argument(
+        "--debug", "-d",
+        action='store_true',
+        default=False,
+        help="Print debug messages on failure or inapplicability")
+
+    parsed_args = parser.parse_args(args)
+
+    rocmlir_gen_flags = ''
+    if 'rocmlir_gen_flags' in parsed_args:
+        rocmlir_gen_flags = parsed_args.rocmlir_gen_flags
+
+    # Impose default behavior when no args have been passed
+    if len(args) == 0:
+        parsed_args.batch_both = True
+
+    confClass = PerfConfiguration
+    opType = Operation.fromName(parsed_args.op)
+    if opType == Operation.CONV:
+        confClass = ConvConfiguration
+    elif opType == Operation.GEMM:
+        confClass = GemmConfiguration
+
+    configs_path = None if parsed_args.config else parsed_args.configs_file
+    paths = perfRunner.create_paths(configs_path, parsed_args.mlir_build_dir, parsed_args.miopen_build_dir)
+    configs = []
+
+    configs = parsed_args.config
+    if not parsed_args.config:
+        if opType == Operation.CONV:
+           configs = perfRunner.getConvConfigurations(paths.configuration_file_path)
+        elif opType == Operation.GEMM:
+            configs = perfRunner.getGemmConfigurations(paths.configuration_file_path)
+        else:
+            raise RuntimeError("Tuning operation was not provided/found")
+
+    if opType == Operation.CONV:
```

Comment:
Ok, yeah, the suggestions above are easier. Just don't have this and pass in a `ProblemConfig` subclass to the benchmarker.